poh-bundle-linux-x64/oh.sh

```diff
@@ -1,68 +1,259 @@
+	cd $POH_PATH/$MYSQL_DIR/
+	echo "Starting MySQL... "
+	$POH_PATH/$MYSQL_DIR/bin/mysqld_safe --defaults-file=$POH_PATH/etc/mysql/my.cnf 2>&1 > /dev/null &
+	if [ $? -ne 0 ]; then
+		echo "Error: Database not started!"
+		exit 2
+	fi
+}
+
+
+# list of arguments expected in the input
+optstring=":hr"
 
-# Find a free port to run MySQL starting from the default port.
-mysql_port=3306
-while [ $(ss -tna | awk '{ print $4 }' | grep ":$mysql_port") ]; do
-	mysql_port=$(expr $mysql_port + 1)
+# function to parse input
+while getopts ${optstring} arg; do
+	case ${arg} in
+	"h")
+		script_usage
+		;;
+	"r")
+        	echo "Resetting Portable Open Hospital installation...."
+		restore_db
+		;;
+	?)
+		echo "Invalid option: -${OPTARG}."
+		exit 2
+		;;
+	esac
 done
 
+
+######## Script start
+
+echo "Checking for software...."
+
+# MySQL
+if [ ! -d "$POH_PATH/$MYSQL_DIR" ]; then
+
+	if [ ! -f "$POH_PATH/$MYSQL_DIR.tar.gz" ]; then
+
+		echo "Warning - MySQL  not found. Do you want to download it ? (-630 Mb)"
+
+		read -p "(y/n)?" choice
+		case "$choice" in 
+			y|Y ) echo "yes";;
+			n|N ) echo "Exiting..."; exit 1;;
+			* ) echo "Invalid choice";;
+		esac
+
+		# Downloading mysql binary
+		echo "Downloading $MYSQL_DIR..."
+		wget $MYSQL_URL/$MYSQL_DIR.tar.gz
+		echo "Unpacking $MYSQL_DIR..."
+		tar zxvf $MYSQL_DIR.tar.gz
+		echo "Done !"
+	fi
+	echo "Using $MYSQL_DIR"
+fi
+
+# Java
+if [ ! -d "$POH_PATH/$JAVA_DIR" ]; then
+
+	if [ ! -f "$POH_PATH/$JAVA_DISTRO.tar.gz" ]; then
+
+		echo "Warning - JAVA  not found. Do you want to download it ? (-180 Mb)"
+
+		read -p "(y/n)?" choice
+		case "$choice" in 
+			y|Y ) echo "yes";;
+			n|N ) echo "Exiting..."; exit 1;;
+			* ) echo "Invalid choice";;
+		esac
+
+		# Downloading openjdk binaries
+		echo "Downloading $JAVA_DISTRO..."
+		wget $JAVA_URL/$JAVA_DISTRO.tar.gz
+		echo "Unpacking $JAVA_DISTRO..."
+		tar zxvf $JAVA_DISTRO.tar.gz
+		echo "Done !"
+	fi
+fi
+
+######## Environment setup
+
+echo "Setting up environment...."
+
 POH_PATH_ESCAPED=$(echo $POH_PATH | sed -e 's/\//\\\//g')
+
+######## DICOM setup
+
 DICOM_MAX_SIZE=$(grep -i '^dicom.max.size' $POH_PATH/$OH_DIR/rsc/dicom.properties.ori  | cut -f2 -d'=')
 : ${DICOM_MAX_SIZE:=$DICOM_DEFAULT_SIZE}
 
+######## Database setup
+
+# Find a free TCP port to run MySQL starting from the default port
+
+echo "Looking for a free TCP port for MySQL database..."
+while [ $(ss -tna | awk '{ print $4 }' | grep ":$MYSQL_PORT") ]; do
+	MYSQL_PORT=$(expr $MYSQL_PORT + 1)
+done
+echo "Found TCP port $MYSQL_PORT!"
+
+# Creating MySQL configuration
+
 rm -f $POH_PATH/etc/mysql/my.cnf or true
-sed -e "s/DICOM_SIZE/$DICOM_MAX_SIZE/g" -e "s/OH_PATH_SUBSTITUTE/$POH_PATH_ESCAPED/g" -e "s/MYSQL_PORT/$mysql_port/" $POH_PATH/etc/mysql/my.ori > $POH_PATH/etc/mysql/my.cnf
+sed -e "s/DICOM_SIZE/$DICOM_MAX_SIZE/g" -e "s/OH_PATH_SUBSTITUTE/$POH_PATH_ESCAPED/g" -e "s/MYSQL_PORT/$MYSQL_PORT/" -e "s/MYSQL_DISTRO/$MYSQL_DIR/g" $POH_PATH/etc/mysql/my.ori > $POH_PATH/etc/mysql/my.cnf
```

Comment:
hasn't `my.ori` been renamed to `my.cnf.dist`?

Reply:
Yes, I'm sorry but I'm not sure if I included correctly the rename file request (still learning github :-); i will update the script and let me know if I have to something more.
(I woul also like propose later on the creation of a "sql" subdir in the structure containig the database.sql file, updating the oh.sh script accordingly)